networking_cisco/plugins/cisco/cfg_agent/device_drivers/asr1k/aci_asr1k_routing_driver.py

```diff
@@ -542,3 +538,38 @@ def _set_vrf_pid(self, ri, port, config):
         # TODO(tbachman): fix to ensure PID uniqueness
         pid = int(hashlib.md5(vrf).hexdigest()[:4], 16)
         return str(pid)
+
+    def _get_snat_pool_name(self, subnet):
+        is_multi_region_enabled = cfg.CONF.multi_region.enable_multi_region
+        if is_multi_region_enabled:
+            region_id = cfg.CONF.multi_region.region_id
+            snat_id = "%s-%s_nat_pool" % (subnet['id'][:self.DEV_NAME_LEN - 6],
+                region_id)
+        else:
+            snat_id = "%s_nat_pool" % (subnet['id'][:self.DEV_NAME_LEN - 6])
+        return snat_id
+
+    def _add_rid_to_snat_list(self, ri, ext_gw_port, subnet):
+        net = netaddr.IPNetwork(subnet['cidr'])
+        snat_id = self._get_snat_pool_name(subnet)
+        if not self._router_ids_by_snat_id.get(snat_id):
+            LOG.debug("++ CREATING SNAT POOL %s" % snat_id)
+            self._do_set_snat_pool(snat_id, subnet['ip'],
+                                   subnet['ip'], str(net.netmask), False)
+        self._router_ids_by_snat_id.setdefault(snat_id, set()).add(
+            ri.router['id'])
+
+    def _remove_rid_from_snat_list(self, ri, ext_gw_port, subnet):
+        net = netaddr.IPNetwork(subnet['cidr'])
+        snat_id = self._get_snat_pool_name(subnet)
+        if self._router_ids_by_snat_id.get(snat_id) and (
+                ri.router['id'] in self._router_ids_by_snat_id[snat_id]):
+            self._router_ids_by_snat_id[snat_id].remove(ri.router['id'])
+        # If htis is the last router in a VRF, then we can safely
+        # delete the VRF from the router config (handled by the driver)
+        if not self._router_ids_by_snat_id.get(snat_id):
```

Comment:
similar to the comments on the previous patch. This if block should be moved under the if block at line 565, then u can remove the if check at line 574.